File: java/core/src/main/java/com/github/fernthedev/lightchat/core/api/plugin/exception/IllegalPluginAccessException.java
package com.github.fernthedev.lightchat.core.api.plugin.exception;

/**
 * Thrown when a com.github.fernthedev.client.plugin attempts to interact with the server when it is not
 * enabled
 */
@SuppressWarnings("serial")
public class IllegalPluginAccessException extends RuntimeException {

    /**
     * Creates a new instance of <code>IllegalPluginAccessException</code>
     * without detail message.
     */
    public IllegalPluginAccessException() {}

    /**
     * Constructs an instance of <code>IllegalPluginAccessException</code>
     * with the specified detail message.
     *
     * @param msg the detail message.
     */
    public IllegalPluginAccessException(String msg) {
        super(msg);
    }
}

File: java/core/src/main/java/com/github/fernthedev/lightchat/core/api/plugin/exception/IllegalPluginAccessException.kt
package com.github.fernthedev.lightchat.core.api.plugin.exception

/**
 * Thrown when a com.github.fernthedev.client.plugin attempts to interact with the server when it is not
 * enabled
 */
class IllegalPluginAccessException : RuntimeException {
    /**
     * Creates a new instance of `IllegalPluginAccessException`
     * without detail message.
     */
    constructor()

    /**
     * Constructs an instance of `IllegalPluginAccessException`
     * with the specified detail message.
     *
     * @param msg the detail message.
     */
    constructor(msg: String?) : super(msg)
}